src/Microsoft.OData.Core/MultipartMixed/ODataMultipartMixedBatchWriter.cs

```diff
@@ -0,0 +1,413 @@
+            return this.RawOutputContext.FlushBuffersAsync()
+                .FollowOnSuccessWith(task => this.DisposeBatchWriterAndSetContentStreamRequestedState());
+        }
+#endif
+
+        /// <summary>
+        /// This method is called to notify that the content stream of a batch operation has been disposed.
+        /// </summary>
+        public override void BatchOperationContentStreamDisposed()
+        {
+            Debug.Assert(this.CurrentOperationMessage != null, "Expected non-null operation message!");
+
+            this.SetState(BatchWriterState.OperationStreamDisposed);
+            this.CurrentOperationRequestMessage = null;
+            this.CurrentOperationResponseMessage = null;
+            this.RawOutputContext.InitializeRawValueWriter();
+        }
+
+        /// <summary>
+        /// This method notifies the listener, that an in-stream error is to be written.
+        /// </summary>
+        /// <remarks>
+        /// This listener can choose to fail, if the currently written payload doesn't support in-stream error at this position.
+        /// If the listener returns, the writer should not allow any more writing, since the in-stream error is the last thing in the payload.
+        /// </remarks>
+        public override void OnInStreamError()
+        {
+            this.RawOutputContext.VerifyNotDisposed();
+            this.SetState(BatchWriterState.Error);
+            this.RawOutputContext.TextWriter.Flush();
+
+            // The OData protocol spec did not defined the behavior when an exception is encountered outside of a batch operation. The batch writer
+            // should not allow WriteError in this case. Note that WCF DS Server does serialize the error in XML format when it encounters one outside of a
+            // batch operation.
+            throw new ODataException(Strings.ODataBatchWriter_CannotWriteInStreamErrorForBatch);
+        }
+
+        /// <summary>
+        /// Flush the output.
+        /// </summary>
+        protected override void FlushSynchronously()
+        {
+            this.RawOutputContext.Flush();
+        }
+
+#if PORTABLELIB
+        /// <summary>
+        /// Flush the output.
+        /// </summary>
+        /// <returns>Task representing the pending flush operation.</returns>
+        protected override Task FlushAsynchronously()
+        {
+            return this.RawOutputContext.FlushAsync();
+        }
+#endif
+
+        /// <summary>
+        /// Starts a new changeset - implementation of the actual functionality.
+        /// </summary>
+        /// <param name="groupId">Ignored value for multipart batch start changeset implementation
+        /// since in this format the changeset boundary is always created randomly for both request and response.</param>
+        protected override void WriteStartChangesetImplementation(/*ignored*/string groupId)

```

Comment:
>/*ignored*/string groupId [](start = 66, length = 25)

don't ignore; pass in a non-null groupId from the base class and pass it on to CreateChangeSetBoundary, rather than generating a new guid there.